### ISSUE TITLE
[help] Parsing dictionaries

### ISSUE THREAD

**Issue by saona-raimundo** [pluots/zspell#71]
I am investigating stems and analyze functions. 
Following [hunspell's documentation on morphological information](https://manpages.ubuntu.com/manpages/lunar/en/man5/hunspell.5.html#:~:text=and%20arrangement%20style.-,optional%20data%20fields,-Default%20%20morphological%20%20and) I came up with this dictionary for "numbers" (actually, only for one digit and zeros).

```
// numbers.dict.
10
0 po:number
1/10 po:number
2/10 po:number
3/10 po:number
4/10 po:number
5/10 po:number
6/10 po:number
7/10 po:number
8/10 po:number
9/10 po:number
```
 ```
// numbers.aff
SFX 10 Y 3
SFX 10 0 0 . is:tens
SFX 10 0 00 . is:hundreds
SFX 10 0 000 . is:thousands
```

The parsing error says
> Error: parse error: parse error: error parsing flag of type 'UTF-8'

Am I doing something wrong? 

The full script I want to run is the following.
```rust
//! Runs with rust-script.
//!
//! ```cargo
//! [dependencies]
//! zspell = "0.4"
//! anyhow = "1.0"
//! ```

use zspell::DictBuilder;

const AFF: &str = r"
SFX 10 Y 3
SFX 10 0 0 . is:tens
SFX 10 0 00 . is:hundreds
SFX 10 0 000 . is:thousands
";
const DICT: &str = r"10
0 po:number
1/10 po:number
2/10 po:number
3/10 po:number
4/10 po:number
5/10 po:number
6/10 po:number
7/10 po:number
8/10 po:number
9/10 po:number
";

fn main() -> anyhow::Result<()> {
	let dict = DictBuilder::new()
	    .config_str(AFF)
	    .dict_str(DICT)
	    .build()?;

	let entry = dict.entry("600");
	let stems: Vec<_> = entry.stems().expect("the word does not have stems.").collect();
	println!("{:?}", stems);

	let morphs: Vec<_> = entry.analyze().expect("the word does not have morphological information.").collect();
	println!("{:?}", morphs);

	Ok(())
}
```

Related to #27

**Comment by tgross35** [comment]
Hey @saona-raimundo thanks for trying it out!

Flags have to be a single character, that is `10` isn't a valid flag for the default flag type of UTF-8. You should either be able to add `FLAG num` to the affix file, or change the flag to a single character.

That error message is pretty bad though, I can improve it

**Comment by tgross35** [comment]
Just as a reference, if you use number flags they are comma separated. Example here: https://github.com/pluots/zspell/blob/52d7febe07a6b12d7f5962a400a1b22bbfcf6625/test-suite/1_pfxsfx_num.test

**Comment by saona-raimundo** [comment]
I saw you already added the example to the test suite, thanks!! :D

**Comment by tgross35** [comment]
Note that it doesn’t yet support unrecognized names, e.g. “number” part of speech wasn’t in the hunspell lists that I’ve seen. I added a fix to allow this but just haven’t released yet

**Comment by saona-raimundo** [comment]
> “number” part of speech wasn’t in the hunspell lists that I’ve seen.

It can perfectly be. 
I am still not familiar with the restrictions in hunspell (like single character suffix flags and what can be a part of speech or not).

**Comment by tgross35** [comment]
That should be fixed with release 0.5.1, flags from the dictionary file and "Other" parts of speech / morph info for any custom info
